zend_abstract_interface/properties/tests/php7-8/properties.cc

```diff
@@ -0,0 +1,134 @@
+extern "C" {
+#include "zai_sapi/zai_sapi.h"
+#include "properties/properties.h"
+#include "functions/functions.h"
+}
+
+#include <catch2/catch.hpp>
+#include <cstring>
+
+#define TEST(name, code) TEST_CASE(name, "[zai properties]") { \
+        REQUIRE(zai_sapi_spinup()); \
+        ZAI_SAPI_TSRMLS_FETCH(); \
+        ZAI_SAPI_ABORT_ON_BAILOUT_OPEN() \
+        REQUIRE(zai_sapi_execute_script("./stubs/classes.php")); \
+        { code } \
+        ZAI_SAPI_ABORT_ON_BAILOUT_CLOSE() \
+        zai_sapi_spindown(); \
+    }
+
+static inline zval *zai_read_property_direct_literal(zend_class_entry *ce, zend_object *obj, const char *name, int name_len) {
+    zend_string *name_str = zend_string_init(name, name_len, 0);
+    zval *val = zai_read_property_direct(ce, obj, name_str);
+    zend_string_release(name_str);
+    return val;
+}
+#define zai_read_property_direct_literal(obj, ce, name) zai_read_property_direct_literal(obj, ce, ZEND_STRL(name))
```

Comment:
I think this would be useful to make public in `properties.h` so that that caller doesn't have to worry about making `zend_strings`. I believe most (or all?) of the time we will use this API, it will be with a string literal. WDYT?

Reply:
Well, our _current_ uses do not have it as string literal, but always zend_string *. I don't mind adding it publicly though (at least for future use).

Reply:
(no, I mean, our current uses are actual zend_string * like `ZSTR_MESSAGE` - but will do anyway)

Reply:
Ah yes, gotcha. Thanks for clarifying. Since the known-string API wasn't added until PHP 7.2, I think using a string literal in this case would be a good way to keep the API consistent across all PHP versions. It'll be great to not need so many preprocessor directives in the tracer code. :)